src/components/workspace-utils.js

```diff
@@ -340,6 +342,46 @@ export const WorkspaceTagSelect = props => {
   })
 }
 
+export const WorkspaceStarControl = ({ workspace, starredWorkspaceIds, onUpdate, style }) => {
+  const [updating, setUpdating] = useState(false)
+
+  const { workspace: { workspaceId } } = workspace
+  const isStarred = _.includes(workspaceId, starredWorkspaceIds)
+
+  const toggleStar = _.flow(
+    Utils.withBusyState(setUpdating),
+    withErrorReporting(`Unable to ${isStarred ? 'unstar' : 'star'} workspace`)
+  )(async star => {
+    if (star) {
+      if (_.size(starredWorkspaceIds) < 50) {
+        const updatedWorkspaceIds = _.concat(starredWorkspaceIds, [workspaceId])
+        await Ajax().User.profile.setPreferences({ starredWorkspaces: _.join(',', updatedWorkspaceIds) })
+        authStore.update(_.set('profile.starredWorkspaces', updatedWorkspaceIds))
+      } else {
+        reportError('Unable to star workspace.',
+          'A maximum of 50 workspaces can be starred. Please un-star another workspace before starring this workspace.')
+      }
+    } else {
+      const updatedWorkspaceIds = _.without([workspaceId], starredWorkspaceIds)
+      await Ajax().User.profile.setPreferences({ starredWorkspaces: _.join(',', updatedWorkspaceIds) })
+      authStore.update(_.set('profile.starredWorkspaces', updatedWorkspaceIds))
+    }
+    onUpdate(star)
+  })
+
+  return h(Interactive, {
```

Comment:
Should this button be disabled and show a tooltip if there are 50 workspaces starred?

Reply:
Done in 67a0acd5ceb98455fdd74f399aabdbf6ff917aaf